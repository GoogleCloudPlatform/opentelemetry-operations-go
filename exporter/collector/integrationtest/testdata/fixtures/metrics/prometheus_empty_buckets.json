{
  "resourceMetrics": [
    {
      "resource": {
        "attributes": [
          {
            "key": "service.name",
            "value": {
              "stringValue": "otel-collector"
            }
          },
          {
            "key": "service.instance.id",
            "value": {
              "stringValue": "0.0.0.0:2112"
            }
          },
          {
            "key": "net.host.port",
            "value": {
              "stringValue": "2112"
            }
          },
          {
            "key": "http.scheme",
            "value": {
              "stringValue": "http"
            }
          }
        ]
      },
      "scopeMetrics": [
        {
          "scope": {},
          "metrics": [
            {
              "name": "kv_prober_write_latency",
              "description": "Latency of successful KV write probes",
              "histogram": {
                "dataPoints": [
                  {
                    "attributes": [],
                    "startTimeUnixNano": "1668799015877000000",
                    "timeUnixNano": "1668799015877000000",
                    "sum": 0,
                    "bucketCounts": [
                      "0"
                    ]
                  }
                ],
                "aggregationTemporality": 2
              }
            },
            {
              "name": "up",
              "description": "The scraping was successful",
              "gauge": {
                "dataPoints": [
                  {
                    "attributes": [],
                    "timeUnixNano": "1668799015877000000",
                    "asDouble": 1
                  }
                ]
              }
            },
            {
              "name": "scrape_duration_seconds",
              "description": "Duration of the scrape",
              "unit": "seconds",
              "gauge": {
                "dataPoints": [
                  {
                    "attributes": [],
                    "timeUnixNano": "1668799015877000000",
                    "asDouble": 0.003655446
                  }
                ]
              }
            },
            {
              "name": "scrape_samples_scraped",
              "description": "The number of samples the target exposed",
              "gauge": {
                "dataPoints": [
                  {
                    "attributes": [],
                    "timeUnixNano": "1668799015877000000",
                    "asDouble": 3
                  }
                ]
              }
            },
            {
              "name": "scrape_samples_post_metric_relabeling",
              "description": "The number of samples remaining after metric relabeling was applied",
              "gauge": {
                "dataPoints": [
                  {
                    "attributes": [],
                    "timeUnixNano": "1668799015877000000",
                    "asDouble": 3
                  }
                ]
              }
            },
            {
              "name": "scrape_series_added",
              "description": "The approximate number of new series in this scrape",
              "gauge": {
                "dataPoints": [
                  {
                    "attributes": [],
                    "timeUnixNano": "1668799015877000000",
                    "asDouble": 3
                  }
                ]
              }
            }
          ]
        }
      ]
    }
  ]
}
