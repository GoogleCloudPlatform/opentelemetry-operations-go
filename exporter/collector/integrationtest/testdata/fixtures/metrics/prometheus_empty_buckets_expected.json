{
  "createTimeSeriesRequests": [
    {
      "name": "projects/fakeprojectid",
      "timeSeries": [
        {
          "metric": {
            "type": "workload.googleapis.com/up",
            "labels": {
              "service_instance_id": "0.0.0.0:2112",
              "service_name": "otel-collector"
            }
          },
          "resource": {
            "type": "generic_task",
            "labels": {
              "job": "otel-collector",
              "location": "global",
              "namespace": "",
              "task_id": "0.0.0.0:2112"
            }
          },
          "metricKind": "GAUGE",
          "valueType": "DOUBLE",
          "points": [
            {
              "interval": {
                "endTime": "1970-01-01T00:00:00Z"
              },
              "value": {
                "doubleValue": 1
              }
            }
          ]
        },
        {
          "metric": {
            "type": "workload.googleapis.com/scrape_duration_seconds",
            "labels": {
              "service_instance_id": "0.0.0.0:2112",
              "service_name": "otel-collector"
            }
          },
          "resource": {
            "type": "generic_task",
            "labels": {
              "job": "otel-collector",
              "location": "global",
              "namespace": "",
              "task_id": "0.0.0.0:2112"
            }
          },
          "metricKind": "GAUGE",
          "valueType": "DOUBLE",
          "points": [
            {
              "interval": {
                "endTime": "1970-01-01T00:00:00Z"
              },
              "value": {
                "doubleValue": 0.003655446
              }
            }
          ],
          "unit": "seconds"
        },
        {
          "metric": {
            "type": "workload.googleapis.com/scrape_samples_scraped",
            "labels": {
              "service_instance_id": "0.0.0.0:2112",
              "service_name": "otel-collector"
            }
          },
          "resource": {
            "type": "generic_task",
            "labels": {
              "job": "otel-collector",
              "location": "global",
              "namespace": "",
              "task_id": "0.0.0.0:2112"
            }
          },
          "metricKind": "GAUGE",
          "valueType": "DOUBLE",
          "points": [
            {
              "interval": {
                "endTime": "1970-01-01T00:00:00Z"
              },
              "value": {
                "doubleValue": 3
              }
            }
          ]
        },
        {
          "metric": {
            "type": "workload.googleapis.com/scrape_samples_post_metric_relabeling",
            "labels": {
              "service_instance_id": "0.0.0.0:2112",
              "service_name": "otel-collector"
            }
          },
          "resource": {
            "type": "generic_task",
            "labels": {
              "job": "otel-collector",
              "location": "global",
              "namespace": "",
              "task_id": "0.0.0.0:2112"
            }
          },
          "metricKind": "GAUGE",
          "valueType": "DOUBLE",
          "points": [
            {
              "interval": {
                "endTime": "1970-01-01T00:00:00Z"
              },
              "value": {
                "doubleValue": 3
              }
            }
          ]
        },
        {
          "metric": {
            "type": "workload.googleapis.com/scrape_series_added",
            "labels": {
              "service_instance_id": "0.0.0.0:2112",
              "service_name": "otel-collector"
            }
          },
          "resource": {
            "type": "generic_task",
            "labels": {
              "job": "otel-collector",
              "location": "global",
              "namespace": "",
              "task_id": "0.0.0.0:2112"
            }
          },
          "metricKind": "GAUGE",
          "valueType": "DOUBLE",
          "points": [
            {
              "interval": {
                "endTime": "1970-01-01T00:00:00Z"
              },
              "value": {
                "doubleValue": 3
              }
            }
          ]
        }
      ]
    }
  ],
  "createMetricDescriptorRequests": [
    {
      "name": "projects/fakeprojectid",
      "metricDescriptor": {
        "name": "kv_prober_write_latency",
        "type": "workload.googleapis.com/kv_prober_write_latency",
        "labels": [
          {
            "key": "service_instance_id"
          },
          {
            "key": "service_name"
          }
        ],
        "metricKind": "CUMULATIVE",
        "valueType": "DISTRIBUTION",
        "description": "Latency of successful KV write probes",
        "displayName": "kv_prober_write_latency"
      }
    },
    {
      "name": "projects/fakeprojectid",
      "metricDescriptor": {
        "name": "scrape_duration_seconds",
        "type": "workload.googleapis.com/scrape_duration_seconds",
        "labels": [
          {
            "key": "service_instance_id"
          },
          {
            "key": "service_name"
          }
        ],
        "metricKind": "GAUGE",
        "valueType": "DOUBLE",
        "unit": "seconds",
        "description": "Duration of the scrape",
        "displayName": "scrape_duration_seconds"
      }
    },
    {
      "name": "projects/fakeprojectid",
      "metricDescriptor": {
        "name": "scrape_samples_post_metric_relabeling",
        "type": "workload.googleapis.com/scrape_samples_post_metric_relabeling",
        "labels": [
          {
            "key": "service_instance_id"
          },
          {
            "key": "service_name"
          }
        ],
        "metricKind": "GAUGE",
        "valueType": "DOUBLE",
        "description": "The number of samples remaining after metric relabeling was applied",
        "displayName": "scrape_samples_post_metric_relabeling"
      }
    },
    {
      "name": "projects/fakeprojectid",
      "metricDescriptor": {
        "name": "scrape_samples_scraped",
        "type": "workload.googleapis.com/scrape_samples_scraped",
        "labels": [
          {
            "key": "service_instance_id"
          },
          {
            "key": "service_name"
          }
        ],
        "metricKind": "GAUGE",
        "valueType": "DOUBLE",
        "description": "The number of samples the target exposed",
        "displayName": "scrape_samples_scraped"
      }
    },
    {
      "name": "projects/fakeprojectid",
      "metricDescriptor": {
        "name": "scrape_series_added",
        "type": "workload.googleapis.com/scrape_series_added",
        "labels": [
          {
            "key": "service_instance_id"
          },
          {
            "key": "service_name"
          }
        ],
        "metricKind": "GAUGE",
        "valueType": "DOUBLE",
        "description": "The approximate number of new series in this scrape",
        "displayName": "scrape_series_added"
      }
    },
    {
      "name": "projects/fakeprojectid",
      "metricDescriptor": {
        "name": "up",
        "type": "workload.googleapis.com/up",
        "labels": [
          {
            "key": "service_instance_id"
          },
          {
            "key": "service_name"
          }
        ],
        "metricKind": "GAUGE",
        "valueType": "DOUBLE",
        "description": "The scraping was successful",
        "displayName": "up"
      }
    }
  ],
  "selfObservabilityMetrics": {
    "createTimeSeriesRequests": [
      {
        "name": "projects/myproject",
        "timeSeries": [
          {
            "metric": {
              "type": "custom.googleapis.com/opencensus/googlecloudmonitoring/point_count",
              "labels": {
                "status": "OK"
              }
            },
            "resource": {
              "type": "global"
            },
            "points": [
              {
                "interval": {
                  "endTime": "1970-01-01T00:00:00Z",
                  "startTime": "1970-01-01T00:00:00Z"
                },
                "value": {
                  "int64Value": "5"
                }
              }
            ]
          },
          {
            "metric": {
              "type": "custom.googleapis.com/opencensus/grpc.io/client/completed_rpcs",
              "labels": {
                "grpc_client_method": "google.monitoring.v3.MetricService/CreateMetricDescriptor",
                "grpc_client_status": "OK"
              }
            },
            "resource": {
              "type": "global"
            },
            "points": [
              {
                "interval": {
                  "endTime": "1970-01-01T00:00:00Z",
                  "startTime": "1970-01-01T00:00:00Z"
                },
                "value": {
                  "int64Value": "6"
                }
              }
            ]
          },
          {
            "metric": {
              "type": "custom.googleapis.com/opencensus/grpc.io/client/completed_rpcs",
              "labels": {
                "grpc_client_method": "google.monitoring.v3.MetricService/CreateTimeSeries",
                "grpc_client_status": "OK"
              }
            },
            "resource": {
              "type": "global"
            },
            "points": [
              {
                "interval": {
                  "endTime": "1970-01-01T00:00:00Z",
                  "startTime": "1970-01-01T00:00:00Z"
                },
                "value": {
                  "int64Value": "1"
                }
              }
            ]
          },
          {
            "metric": {
              "type": "custom.googleapis.com/opencensus/grpc.io/client/received_bytes_per_rpc",
              "labels": {
                "grpc_client_method": "google.monitoring.v3.MetricService/CreateMetricDescriptor"
              }
            },
            "resource": {
              "type": "global"
            },
            "points": [
              {
                "interval": {
                  "endTime": "1970-01-01T00:00:00Z",
                  "startTime": "1970-01-01T00:00:00Z"
                },
                "value": {
                  "distributionValue": {
                    "bucketOptions": {
                      "explicitBuckets": {
                        "bounds": [
                          0,
                          1024,
                          2048,
                          4096,
                          16384,
                          65536,
                          262144,
                          1048576,
                          4194304,
                          16777216,
                          67108864,
                          268435456,
                          1073741824,
                          4294967296
                        ]
                      }
                    },
                    "bucketCounts": [
                      "0",
                      "0",
                      "0",
                      "0",
                      "0",
                      "0",
                      "0",
                      "0",
                      "0",
                      "0",
                      "0",
                      "0",
                      "0",
                      "0",
                      "0"
                    ]
                  }
                }
              }
            ]
          },
          {
            "metric": {
              "type": "custom.googleapis.com/opencensus/grpc.io/client/received_bytes_per_rpc",
              "labels": {
                "grpc_client_method": "google.monitoring.v3.MetricService/CreateTimeSeries"
              }
            },
            "resource": {
              "type": "global"
            },
            "points": [
              {
                "interval": {
                  "endTime": "1970-01-01T00:00:00Z",
                  "startTime": "1970-01-01T00:00:00Z"
                },
                "value": {
                  "distributionValue": {
                    "bucketOptions": {
                      "explicitBuckets": {
                        "bounds": [
                          0,
                          1024,
                          2048,
                          4096,
                          16384,
                          65536,
                          262144,
                          1048576,
                          4194304,
                          16777216,
                          67108864,
                          268435456,
                          1073741824,
                          4294967296
                        ]
                      }
                    },
                    "bucketCounts": [
                      "0",
                      "0",
                      "0",
                      "0",
                      "0",
                      "0",
                      "0",
                      "0",
                      "0",
                      "0",
                      "0",
                      "0",
                      "0",
                      "0",
                      "0"
                    ]
                  }
                }
              }
            ]
          },
          {
            "metric": {
              "type": "custom.googleapis.com/opencensus/grpc.io/client/roundtrip_latency",
              "labels": {
                "grpc_client_method": "google.monitoring.v3.MetricService/CreateMetricDescriptor"
              }
            },
            "resource": {
              "type": "global"
            },
            "points": [
              {
                "interval": {
                  "endTime": "1970-01-01T00:00:00Z",
                  "startTime": "1970-01-01T00:00:00Z"
                },
                "value": {
                  "distributionValue": {
                    "bucketOptions": {
                      "explicitBuckets": {
                        "bounds": [
                          0,
                          0.01,
                          0.05,
                          0.1,
                          0.3,
                          0.6,
                          0.8,
                          1,
                          2,
                          3,
                          4,
                          5,
                          6,
                          8,
                          10,
                          13,
                          16,
                          20,
                          25,
                          30,
                          40,
                          50,
                          65,
                          80,
                          100,
                          130,
                          160,
                          200,
                          250,
                          300,
                          400,
                          500,
                          650,
                          800,
                          1000,
                          2000,
                          5000,
                          10000,
                          20000,
                          50000,
                          100000
                        ]
                      }
                    },
                    "bucketCounts": [
                      "0",
                      "0",
                      "0",
                      "0",
                      "0",
                      "0",
                      "0",
                      "0",
                      "0",
                      "0",
                      "0",
                      "0",
                      "0",
                      "0",
                      "0",
                      "0",
                      "0",
                      "0",
                      "0",
                      "0",
                      "0",
                      "0",
                      "0",
                      "0",
                      "0",
                      "0",
                      "0",
                      "0",
                      "0",
                      "0",
                      "0",
                      "0",
                      "0",
                      "0",
                      "0",
                      "0",
                      "0",
                      "0",
                      "0",
                      "0",
                      "0",
                      "0"
                    ]
                  }
                }
              }
            ]
          },
          {
            "metric": {
              "type": "custom.googleapis.com/opencensus/grpc.io/client/roundtrip_latency",
              "labels": {
                "grpc_client_method": "google.monitoring.v3.MetricService/CreateTimeSeries"
              }
            },
            "resource": {
              "type": "global"
            },
            "points": [
              {
                "interval": {
                  "endTime": "1970-01-01T00:00:00Z",
                  "startTime": "1970-01-01T00:00:00Z"
                },
                "value": {
                  "distributionValue": {
                    "bucketOptions": {
                      "explicitBuckets": {
                        "bounds": [
                          0,
                          0.01,
                          0.05,
                          0.1,
                          0.3,
                          0.6,
                          0.8,
                          1,
                          2,
                          3,
                          4,
                          5,
                          6,
                          8,
                          10,
                          13,
                          16,
                          20,
                          25,
                          30,
                          40,
                          50,
                          65,
                          80,
                          100,
                          130,
                          160,
                          200,
                          250,
                          300,
                          400,
                          500,
                          650,
                          800,
                          1000,
                          2000,
                          5000,
                          10000,
                          20000,
                          50000,
                          100000
                        ]
                      }
                    },
                    "bucketCounts": [
                      "0",
                      "0",
                      "0",
                      "0",
                      "0",
                      "0",
                      "0",
                      "0",
                      "0",
                      "0",
                      "0",
                      "0",
                      "0",
                      "0",
                      "0",
                      "0",
                      "0",
                      "0",
                      "0",
                      "0",
                      "0",
                      "0",
                      "0",
                      "0",
                      "0",
                      "0",
                      "0",
                      "0",
                      "0",
                      "0",
                      "0",
                      "0",
                      "0",
                      "0",
                      "0",
                      "0",
                      "0",
                      "0",
                      "0",
                      "0",
                      "0",
                      "0"
                    ]
                  }
                }
              }
            ]
          },
          {
            "metric": {
              "type": "custom.googleapis.com/opencensus/grpc.io/client/sent_bytes_per_rpc",
              "labels": {
                "grpc_client_method": "google.monitoring.v3.MetricService/CreateMetricDescriptor"
              }
            },
            "resource": {
              "type": "global"
            },
            "points": [
              {
                "interval": {
                  "endTime": "1970-01-01T00:00:00Z",
                  "startTime": "1970-01-01T00:00:00Z"
                },
                "value": {
                  "distributionValue": {
                    "bucketOptions": {
                      "explicitBuckets": {
                        "bounds": [
                          0,
                          1024,
                          2048,
                          4096,
                          16384,
                          65536,
                          262144,
                          1048576,
                          4194304,
                          16777216,
                          67108864,
                          268435456,
                          1073741824,
                          4294967296
                        ]
                      }
                    },
                    "bucketCounts": [
                      "0",
                      "0",
                      "0",
                      "0",
                      "0",
                      "0",
                      "0",
                      "0",
                      "0",
                      "0",
                      "0",
                      "0",
                      "0",
                      "0",
                      "0"
                    ]
                  }
                }
              }
            ]
          },
          {
            "metric": {
              "type": "custom.googleapis.com/opencensus/grpc.io/client/sent_bytes_per_rpc",
              "labels": {
                "grpc_client_method": "google.monitoring.v3.MetricService/CreateTimeSeries"
              }
            },
            "resource": {
              "type": "global"
            },
            "points": [
              {
                "interval": {
                  "endTime": "1970-01-01T00:00:00Z",
                  "startTime": "1970-01-01T00:00:00Z"
                },
                "value": {
                  "distributionValue": {
                    "bucketOptions": {
                      "explicitBuckets": {
                        "bounds": [
                          0,
                          1024,
                          2048,
                          4096,
                          16384,
                          65536,
                          262144,
                          1048576,
                          4194304,
                          16777216,
                          67108864,
                          268435456,
                          1073741824,
                          4294967296
                        ]
                      }
                    },
                    "bucketCounts": [
                      "0",
                      "0",
                      "0",
                      "0",
                      "0",
                      "0",
                      "0",
                      "0",
                      "0",
                      "0",
                      "0",
                      "0",
                      "0",
                      "0",
                      "0"
                    ]
                  }
                }
              }
            ]
          }
        ]
      }
    ],
    "createMetricDescriptorRequests": [
      {
        "name": "projects/myproject",
        "metricDescriptor": {
          "name": "projects/myproject/metricDescriptors/custom.googleapis.com/opencensus/googlecloudmonitoring/point_count",
          "type": "custom.googleapis.com/opencensus/googlecloudmonitoring/point_count",
          "labels": [
            {
              "key": "status"
            }
          ],
          "metricKind": "CUMULATIVE",
          "valueType": "INT64",
          "unit": "1",
          "description": "Count of metric points written to Cloud Monitoring.",
          "displayName": "OpenCensus/googlecloudmonitoring/point_count"
        }
      },
      {
        "name": "projects/myproject",
        "metricDescriptor": {
          "name": "projects/myproject/metricDescriptors/custom.googleapis.com/opencensus/grpc.io/client/completed_rpcs",
          "type": "custom.googleapis.com/opencensus/grpc.io/client/completed_rpcs",
          "labels": [
            {
              "key": "grpc_client_method"
            },
            {
              "key": "grpc_client_status"
            }
          ],
          "metricKind": "CUMULATIVE",
          "valueType": "INT64",
          "unit": "1",
          "description": "Count of RPCs by method and status.",
          "displayName": "OpenCensus/grpc.io/client/completed_rpcs"
        }
      },
      {
        "name": "projects/myproject",
        "metricDescriptor": {
          "name": "projects/myproject/metricDescriptors/custom.googleapis.com/opencensus/grpc.io/client/received_bytes_per_rpc",
          "type": "custom.googleapis.com/opencensus/grpc.io/client/received_bytes_per_rpc",
          "labels": [
            {
              "key": "grpc_client_method"
            }
          ],
          "metricKind": "CUMULATIVE",
          "valueType": "DISTRIBUTION",
          "unit": "By",
          "description": "Distribution of bytes received per RPC, by method.",
          "displayName": "OpenCensus/grpc.io/client/received_bytes_per_rpc"
        }
      },
      {
        "name": "projects/myproject",
        "metricDescriptor": {
          "name": "projects/myproject/metricDescriptors/custom.googleapis.com/opencensus/grpc.io/client/roundtrip_latency",
          "type": "custom.googleapis.com/opencensus/grpc.io/client/roundtrip_latency",
          "labels": [
            {
              "key": "grpc_client_method"
            }
          ],
          "metricKind": "CUMULATIVE",
          "valueType": "DISTRIBUTION",
          "unit": "ms",
          "description": "Distribution of round-trip latency, by method.",
          "displayName": "OpenCensus/grpc.io/client/roundtrip_latency"
        }
      },
      {
        "name": "projects/myproject",
        "metricDescriptor": {
          "name": "projects/myproject/metricDescriptors/custom.googleapis.com/opencensus/grpc.io/client/sent_bytes_per_rpc",
          "type": "custom.googleapis.com/opencensus/grpc.io/client/sent_bytes_per_rpc",
          "labels": [
            {
              "key": "grpc_client_method"
            }
          ],
          "metricKind": "CUMULATIVE",
          "valueType": "DISTRIBUTION",
          "unit": "By",
          "description": "Distribution of bytes sent per RPC, by method.",
          "displayName": "OpenCensus/grpc.io/client/sent_bytes_per_rpc"
        }
      }
    ]
  }
}
